### PR TITLE
CO-3005 modification to controller. 

### PR DESCRIPTION
**Pull request by quentingigon**
A second PR on compassion-switzerland implements second half of fix. When getting the sponsorship using browse, its parameters were then inaccessible

### CODE REVIEW

**Comment by ecino** [comment]
Is it because sponsorship_id is a string? Then, could we just make browse(int(sponsorship_id)) instead? And maybe check that sponsorship_id is not None.

**Comment by ecino** [comment]
https://github.com/CompassionCH/compassion-switzerland/pull/1004

**Comment by quentingigon** [comment]
I will try it

**Comment by quentingigon** [comment]
> Is it because sponsorship_id is a string? Then, could we just make browse(int(sponsorship_id)) instead?

No, I tried all possibilities and with browse(), I had no access to its parameters

File: sms_sponsorship/controllers/main.py [hunk 1]
```diff
@@ -142,20 +142,24 @@ def sms_change_child(self, child_request_id):
...
+                return self.make_response(
+                    'recurring.contract',
+                    model_id=sponsorship and sponsorship.id,
+                    **kwargs
+                )
+            else:
+                raise NotFound()
         else:
-            raise NotFound()
+            raise Exception("Please provide a sponsorship_id")
```
Comment (ecino) [inline sms_sponsorship/controllers/main.py]: raise NotFound() was better (for user interface), however you can add a line to log an error in the logs
```
import logging

_logger = logging.getLogger(__name__)

_logger.error("No sponsorship_id was given to the route")

File: sms_sponsorship/controllers/main.py [hunk 2]
```diff
@@ -142,20 +142,24 @@ def sms_change_child(self, child_request_id):
...
+                # Sponsorship is already confirmed
+                return self.sms_registration_confirmation(sponsorship.id, **kwargs)
+            if sponsorship.state == "draft":
+                return self.make_response(
+                    'recurring.contract',
+                    model_id=sponsorship and sponsorship.id,
+                    **kwargs
+                )
+            else:
+                raise NotFound()
```
Comment (ecino) [inline sms_sponsorship/controllers/main.py:165]: Add a log
```
_logger.error("No valid sponsorship found for given id : %s", sponsorship_id)
```

**Review by ecino — CHANGES_REQUESTED** [review]

**Review by ecino — APPROVED** [review]

**Pull request merged**